### PR TITLE
bridge-pangoro-chapel

### PR DESCRIPTION
**Pull request by furoxr**

### CODE REVIEW

**Comment by fewensa** [comment]
There should use pangolin chain

**Comment by furoxr** [comment]
CC @fewensa, 
1. Before running the service, there should be a config file named bridge-pangoro-chapel.toml under ~/.bridger. Like this:
``` toml
[pangoro]
endpoint = "wss://pangoro-rpc.darwinia.network:443"
relayer_private_key = "//Alice"
# relayer_real_account = "0x..."

[chapel]
endpoint = "https://data-seed-prebsc-1-s1.binance.org:8545/"
```
2. In the directory of bridges/pangoro-chapel/
```
cargo run start
```
image

**Comment by HackFisher** [comment]
@furoxr Is this ready for review?

**Comment by furoxr** [comment]
> @furoxr Is this ready for review?

Yes, the header relaying from chapel to pangoro has been finished, and is ready for review.

**Comment by HackFisher** [comment]
@furoxr The CI failed.

**Comment by fewensa** [comment]
> ```toml
> [pangoro]
> endpoint = "wss://pangoro-rpc.darwinia.network:443"
> relayer_private_key = "//Alice"
> # relayer_real_account = "0x..."
> 
> [chapel]
> endpoint = "https://data-seed-prebsc-1-s1.binance.org:8545/"
> ```


Please put this config file to `.maintain/config/bridge-pangoro-chapel.toml` as template config file.

**Review by HackFisher — APPROVED** [review]

File: bridges/pangoro-chapel/bridge/src/command/types.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+use structopt::StructOpt;
+
+/// Bridge template options
+#[derive(Debug, StructOpt)]
+#[structopt(name = "bridge-template", about = "Bridge template")]
```
Comment (fewensa) [inline bridges/pangoro-chapel/bridge/src/command/types.rs:5]:
```suggestion
#[structopt(name = "pangoro-chapel", about = "Bridge pangoro-chapel")]
```

File: frame/assistants/pangoro-subxt/scripts/codegen.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
```
Comment (fewensa) [inline frame/assistants/pangoro-subxt/scripts/codegen.sh:1]: Remove this script/dirctory and use `scripts/subxt-gen.sh` script, if that script file does not have to execute permission please set it.

File: bridges/pangoro-chapel/bridge/src/service/header_relay.rs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+impl Service for HeaderRelayService {
+    type Bus = PangoroChapelBus;
+    type Lifeline = color_eyre::Result<Self>;
+
+    fn spawn(_bus: &Self::Bus) -> Self::Lifeline {
+        tracing::trace!("Spawn service HeaderRelayService");
+        let config: PangoroChapelConfig = Config::restore(Names::BridgePangoroChapel)?;
+
+        println!("{:?}", config);
```
Comment (fewensa) [inline bridges/pangoro-chapel/bridge/src/service/header_relay.rs:30]: Please remove this or use tracing

**Review by fewensa — APPROVED** [review]

**Pull request merged**